nxcomp/Loop.cpp

```diff
@@ -9340,6 +9341,100 @@ int ParseCommandLineOptions(int argc, const char **argv)
 
           return -1;
         }
+        case 'd':
+        {
+          if ( argi+1 >= argc )
+          {
+            PrintUsageInfo(nextArg, 0);  
+            return -1;
+          }
+
+          int level = 0;
+	  level = strtol(argv[argi+1], NULL, 10);
```

Comment:
Reset `errno` to 0 before calling `strtol()`. Succeeding calls might or might not set `errno`, so if `strtol()` succeeds you might not know if the value in `errno` is stale.